### PR TITLE
[WIP] Exposes `SESSION_ENGINE` as ENVVAR Setting, Always Run `collectstatic`

### PR DESCRIPTION
**Pull request by gneville-ot**
# Closes: #2477
# What's Changed

- Changes the Docker Entrypoint so it runs `nautobot-server collectstatic` if `NAUTOBOT_DOCKER_SKIP_INIT=True`, this is needed so static content can be served when standing up in maintenance mode.
- Allows for easier overriding of the SESSION_ENGINE default with ENVVAR `NAUTOBOT_SESSION_ENGINE`, this is changed when standing up in maintenance mode.

# TODO

- [X] Explanation of Change(s)
- [X] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [X] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Comment by jathanism** [comment]
Thanks for this submission! Sorry we didn't get to it for the 1.4.~6~7 release.

File: nautobot/core/templates/nautobot_config.py.j2 [hunk 1]
```diff
@@ -224,6 +224,10 @@ PLUGINS = []
 REMOTE_AUTH_AUTO_CREATE_USER = False
 REMOTE_AUTH_HEADER = "HTTP_REMOTE_USER"
 
+# Set the SESSION_ENGINE
+# By default, Nautobot will use 'django.contrib.sessions.backends.db' however if you need a maintenance mode / read-only instance then this can be overriden
+SESSION_ENGINE = os.getenv("NAUTOBOT_SESSION_ENGINE", "django.contrib.sessions.backends.db")
+
```
Comment (glennmatthews) [inline nautobot/core/templates/nautobot_config.py.j2:230]: Can this be implemented in `nautobot/core/settings.py` instead? (See also #2578)

**Review by glennmatthews — COMMENTED** [review]
Thank you!

**Comment by gneville-ot** [comment]
Hi @jathanism, @glennmatthews ,

I've moved the setting in to `nautobot/core/settings.py` and tested and still works.

Thanks

**Comment by jathanism** [comment]
@nniehoff Can you please give this a look!

**Review by nniehoff — CHANGES_REQUESTED** [review]
This makes sense in this case, however we definitely have use cases we need to account for where we don't want to run collectstatic either.  We need to account for these.

**Comment by gneville-ot** [comment]
Hi @nniehoff,

As a regular user of nautobot I won't know all of these use cases. 
As I see it now, `collectstatic` always runs if you're spinning up with docker and doesn't if you set `NAUTOBOT_DOCKER_SKIP_INIT=True`. Is there any reason not to run `collectstatic` when setting  `NAUTOBOT_DOCKER_SKIP_INIT=True`?

I won't be able to progress this PR without feedback from you.

**Comment by nniehoff** [comment]
@gneville-ot that's exactly what I am saying there are use cases out there where when `NAUTOBOT_DOCKER_SKIP_INIT=True` we truly do mean skip all initialization to include `collectstatic`.  Mainly these are around where we use shared file systems for static such as S3.  Nearly all of the HA deployments we do are active/active deployments of Nautobot with active/active HA postgres behind them.  I'm sorry for the trailblazing you are having to do with this issue.  For this PR I'd like to see some other options maybe if `NAUTOBOT_DOCKER_SKIP_INIT=True`  we have a number of other variables to address all of the components of `post_upgrade` which can be found [here](https://github.com/nautobot/nautobot/blob/develop/nautobot/core/management/commands/post_upgrade.py#L76-L114)

**Review by glennmatthews — CHANGES_REQUESTED** [review]
Nick's concerns need to be addressed. I agree that we probably need to keep `NAUTOBOT_DOCKER_SKIP_INIT` as meaning "skip all init tasks" and instead add a different (set of) variables that get passed through to the `nautobot-server post_upgrade` command so that we can for example run `post_upgrade --no-migrate --no-clearsessions --no-remove-stale-contenttypes --no-trace-paths --no-invalidate-all` if it's desirable to only run `collectstatic`... looking at that complexity, maybe we could consider adding a `--only` option to the `post_upgrade` command (`nautobot-server post_upgrade --only collectstatic`)?

**Comment by bryanculver** [comment]
Pending feedback: https://github.com/nautobot/nautobot/issues/2477#issuecomment-1329703574

**Comment by bryanculver** [comment]
Closing as this largely needs to be broken out into its own two issues that this doesn't fully address:
- We have a larger need around what maintenance mode does and does not support, see #3183
- Based on @glennmatthews' comment above we could have a few more knobs for the Docker entryfile to tune what commands get run on startup instead of an all-or-nothing post_upgrade. The use case here is just one of several but there is no simple way to "override" the entryfile without drift management, see #3733

**Pull request closed without merging**